lib/Runner.js

```diff
@@ -71,133 +47,87 @@ function getIndirectDeps(projectRootDir /*: string */) /*: Promise<Object> */ {
 
-    proc.on('close', function (code) {
-      if (stderrStr !== '') {
-        reject(stderrStr);
-      } else if (code !== 0) {
-        reject('Finding test interfaces failed, exiting with code ' + code);
-      }
+${
+  isPackageProject
+    ? 'Move it to tests/ or src/ in your project root.'
+    : 'Move it to tests/ in your project root, or make sure it is covered by "source-directories" in your elm.json.'
+}
+  `.trim();
+}
 
-      let modules;
+function multipleSourceDirectoriesError(filePath, matchingSourceDirs) {
+  return `
+This file:
 
-      try {
-        modules = JSON.parse(jsonStr).testModules;
-      } catch (err) {
-        reject('Received invalid JSON from test interface search: ' + err);
-        return;
-      }
+${filePath}
 
-      var filteredModules = filterMap(modules, function (mod) {
-        // Must have at least 1 value of type Test. Otherwise ignore this module.
-        const normalizedPath = path.resolve(path.join(testRootDir, mod.path));
-        if (mod.tests.length > 0 && testFilePaths.includes(normalizedPath)) {
-          return {
-            moduleName: mod.moduleName,
-            path: normalizedPath,
-            tests: mod.tests,
-          };
-        } else {
-          return undefined;
-        }
-      });
+…matches more than one source directory:
 
-      return verifyModules(testFilePaths)
-        .then(function () {
-          resolve(filteredModules);
-        })
-        .catch(reject);
-    });
-  });
-}
+${matchingSourceDirs.join('\n')}
 
-// Check for modules where the name doesn't match the filename.
-// elm-make won't get a chance to detect this; they'll be filtered out first.
-function verifyModules(filePaths) {
-  return Promise.all(
-    filePaths.map(function (filePath) {
-      return getFirstLine(filePath).then(function (line) {
-        var matches = line.match(/^(?:(?:port|effect)\s+)?module\s+(\S+)\s*/);
-
-        if (matches) {
-          var moduleName = matches[1];
-          var testModulePaths = moduleFromTestName(moduleName);
-          var modulePath = moduleFromFilePath(filePath);
-
-          // A module path matches if it lines up completely with a known one.
-          if (
-            !testModulePaths.every(function (testModulePath, index) {
-              return testModulePath === modulePath[index];
-            })
-          ) {
-            return Promise.reject(
-              filePath +
-                ' has a module declaration of "' +
-                moduleName +
-                '" - which does not match its filename!'
-            );
-          }
-        } else {
-          return Promise.reject(
-            filePath +
-              ' has an invalid module declaration. Check the first line of the file and make sure it has a valid module declaration there!'
-          );
-        }
-      });
-    })
-  );
+Edit "source-directories" in your elm.json and try to make it so no source directory contains another source directory!
```

Comment:
This message is inspired by Elm’s error message when you have two similar source directories.